synapse/storage/databases/main/monthly_active_users.py

```diff
@@ -354,3 +354,17 @@ async def populate_monthly_active_users(self, user_id):
                         await self.upsert_monthly_active_user(user_id)
             elif now - last_seen_timestamp > LAST_SEEN_GRANULARITY:
                 await self.upsert_monthly_active_user(user_id)
+
+    async def remove_deactivated_user(self, user_id: str) -> None:
```

Comment:
Sorry, should have said this yesterday: I think this name should make it clear we're only deleting from the `monthly_active_users` table and not e.g. from the `users` table.